### PR TITLE
nctalk: support message deletion

### PR DESCRIPTION
**Pull request by gary-kim**
Support forwarding and receiving message deletions.
Note: Nextcloud Talk only allows deleting messages if the message was sent less than 6 hours ago or the user is a moderator in the room.

### CODE REVIEW

**Comment by gary-kim** [comment]
It might be about time to start splitting this bridge out into a `handlers.go` file.

**Comment by gary-kim** [comment]
@42wim This is ready for you to take a look :)

File: bridge/nctalk/nctalk.go [hunk 1]
```diff
@@ -208,6 +204,55 @@ func (b *Btalk) handleSendingFile(msg *config.Message, r *Broom) error {
...
+	}
+
+	b.Log.Debugf("<= Message is %#v", remoteMessage)
+	b.Remote <- remoteMessage
+}
+
+func (b *Btalk) handleDeletingMessage(msg *ocs.TalkRoomMessageData, r *Broom) {
+	remoteMessage := config.Message{
+		Event:    config.EventMsgDelete,
+		Text:     formatRichObjectString(msg.Message, msg.MessageParameters),
```
Comment (42wim) [inline bridge/nctalk/nctalk.go:236]: To be consistent with the other bridges make this also `config.EventMsgDelete` makes it also easier to spot in debugs.

File: bridge/nctalk/nctalk.go [hunk 2]
```diff
@@ -208,6 +204,55 @@ func (b *Btalk) handleSendingFile(msg *config.Message, r *Broom) error {
...
+		Channel:  r.room.Token,
+		Username: DisplayName(msg, b.guestSuffix()),
+		UserID:   msg.ActorID,
```
Comment (42wim) [inline bridge/nctalk/nctalk.go:239]: Those are not really relevant for a deletemessage, can be removed.

**Review by 42wim — COMMENTED** [review]
Again small nitpicks :)

File: bridge/nctalk/nctalk.go (hunk 2 above)
Comment (gary-kim) [reply bridge/nctalk/nctalk.go]: Done

File: bridge/nctalk/nctalk.go (hunk 1 above)
Comment (gary-kim) [reply bridge/nctalk/nctalk.go]: Done

**Review by 42wim — APPROVED** [review]

**Comment by 42wim** [comment]
🎉

**Pull request merged**